lib/subordinateio.h

```diff
@@ -43,6 +43,9 @@ extern int sub_gid_unlock (void);
 extern int sub_gid_add (const char *owner, gid_t start, unsigned long count);
 extern int sub_gid_remove (const char *owner, gid_t start, unsigned long count);
 extern uid_t sub_gid_find_free_range(gid_t min, gid_t max, unsigned long count);
+
+extern void free_subid_pointer(void *ptr);
+
```

Comment:
Now that we have a deallocator, should we use `[[gnu::malloc(deallocator)]]` for documenting which functions should be freed by this one?

Reply:
I didn't quite get it. According to [GCC manuals][1], this attribute can be applied only to `malloc`-like functions (functions that returns a pointer with signature `<type> *(*)(...)`). I have checked empirically that GCC indeed generates a warning with the snippet below

```cpp
/* lib/subordinate.io.h */
[[gnu::malloc(free_subid_pointer)]]
extern int list_owner_ranges(..., struct subid_range **ranges);
```

and produces the following.

```
../lib/subordinateio.h:33:81: warning: ‘malloc’ attribute ignored on functions returning ‘int’; valid only for pointer return types [-Wattributes]
```

[1]: https://gcc.gnu.org/onlinedocs/gcc/Common-Function-Attributes.html#index-malloc-function-attribute

Reply:
Ahhh, we pass the allocated pointer by reference.  Meh.